halo2_proofs/src/plonk/evaluation.rs

```diff
@@ -534,41 +662,45 @@ impl<C: CurveAffine> Evaluator<C> {
-                                        - product_coset[idx])
-                                        * l_last[idx]);
-                                // (1 - (l_last(X) + l_blind(X))) * (
-                                //   z(\omega X) (a'(X) + \beta) (s'(X) + \gamma)
-                                //   - z(X) (\theta^{m-1} a_0(X) + ... + a_{m-1}(X) + \beta)
-                                //          (\theta^{m-1} s_0(X) + ... + s_{m-1}(X) + \gamma)
-                                // ) = 0
-                                *value = *value * y
-                                    + ((product_coset[r_next]
-                                        * (permuted_input_coset[idx] + beta)
-                                        * (permuted_table_coset[idx] + gamma)
-                                        - product_coset[idx] * table_value)
-                                        * l_active_row[idx]);
-                                // Check that the first values in the permuted input expression and permuted
-                                // fixed expression are the same.
-                                // l_0(X) * (a'(X) - s'(X)) = 0
-                                *value = *value * y + (a_minus_s * l0[idx]);
-                                // Check that each value in the permuted lookup input expression is either
-                                // equal to the value above it, or the value at the same index in the
-                                // permuted table expression.
-                                // (1 - (l_last + l_blind)) * (a′(X) − s′(X))⋅(a′(X) − a′(\omega^{-1} X)) = 0
-                                *value = *value * y
-                                    + (a_minus_s
-                                        * (permuted_input_coset[idx]
-                                            - permuted_input_coset[r_prev])
-                                        * l_active_row[idx]);
+                                let lhs = {
+                                    // τ(X) * Π(φ_i(X)) * (ϕ(gX) - ϕ(X))
+                                    table_value * inputs_prod * (phi_coset[r_next] - phi_coset[idx])
+                                };
+
+                                #[cfg(feature = "logup_skip_inv")]
+                                let rhs = {
+                                    //   τ(X) * Π(φ_i(X)) * (∑ 1/(φ_i(X)) - m(X) / τ(X))))
+                                    // = ∑_i τ(X) * Π_{j != i} φ_j(X) - m(X) * Π(φ_i(X))
+                                    let inputs = (0..inputs_value.len())
+                                        .map(|i| {
+                                            inputs_value
+                                                .iter()
+                                                .enumerate()
+                                                .filter(|(j, _)| *j != i)
+                                                .fold(C::Scalar::ZERO, |acc, (_, x)| acc * *x)
```

Comment:
It looks weird as the result would always be 0. Perhaps, 1 instead of 0 was intended?

Reply:
Nice catch! This bug is fixed in https://github.com/scroll-tech/halo2/pull/83.